### PR TITLE
Update GUI for Template form

### PR DESCRIPTION
**Pull request by Beacze**
- change GUI for template form
- add active flag to Template
- change relationships between process and template

Note: it will have merge conflicts with #1678

### CODE REVIEW

File: Kitodo-DataManagement/src/main/resources/db/migration/V2_41__Change_project_template_to_many.sql [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+-- 5. Add foreign keys to cross table
+
+ALTER TABLE project_x_template ENGINE=InnoDB;
+ALTER TABLE project_x_template
+   ADD CONSTRAINT `FK_project_x_template_project_id`
+ FOREIGN KEY (project_id) REFERENCES project (id);
+ALTER TABLE project_x_template
+   ADD CONSTRAINT `FK_project_x_template_template_id`
+ FOREIGN KEY (template_id) REFERENCES template (id);
```
Comment (henning-gerhardt) [inline Kitodo-DataManagement/src/main/resources/db/migration/V2_41__Change_project_template_to_many.sql]: Missing new line at end of file.

File: Kitodo-DataManagement/src/main/resources/db/migration/V2_41__Change_project_template_to_many.sql [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+ALTER TABLE template DROP FOREIGN KEY `FK_template_project_id`;
+
+-- 4. Remove process column
+
+ALTER TABLE template DROP COLUMN project_id;
+
+-- 5. Add foreign keys to cross table
+
+ALTER TABLE project_x_template ENGINE=InnoDB;
```
Comment (henning-gerhardt) [inline Kitodo-DataManagement/src/main/resources/db/migration/V2_41__Change_project_template_to_many.sql]: You can move this to creating of table.

File: Kitodo/src/main/java/de/sub/goobi/forms/MassImportForm.java [hunk 3]
```diff
@@ -104,9 +106,10 @@ public MassImportForm() {
      *
      * @return String
      */
-    public String prepare(int id) {
+    public String prepare(int templateId, int projectId) {
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/de/sub/goobi/forms/MassImportForm.java]: As you change the method signature, adjust the JavaDoc.

File: Kitodo/src/main/java/de/sub/goobi/forms/ProzesskopieForm.java [hunk 4]
```diff
@@ -283,13 +282,13 @@ public void selectClick() {
      *
      * @return empty String
      */
-    public String prepare(int id) {
+    public String prepare(int templateId, int projectId) {
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/de/sub/goobi/forms/ProzesskopieForm.java]: As you change the method signature, adjust the JavaDoc.

File: Kitodo-DataManagement/src/main/resources/db/migration/V2_41__Change_project_template_to_many.sql (hunk 1 above)
Comment (Beacze) [reply Kitodo-DataManagement/src/main/resources/db/migration/V2_41__Change_project_template_to_many.sql]: Added

**Comment by Beacze** [comment]
Naming must be resolved after one of PR is merged. We don't know yet which one will be merged as first.

**Comment by henning-gerhardt** [comment]
There is at least one merge conflict in

- Kitodo/src/main/webapp/WEB-INF/templates/includes/projects/templateList.xhtml

Please fix / solve this conflict.

File: Kitodo/src/test/java/org/kitodo/selenium/testframework/Browser.java [hunk 5]
```diff
@@ -195,6 +195,7 @@ public static File captureScreenShot() {
         List<String> data = new ArrayList<>();
         for (WebElement row : rows) {
             data.add(getCellDataByRow(row, columnIndex));
+            System.out.println(getCellDataByRow(row, columnIndex));
```
Comment (henning-gerhardt) [inline Kitodo/src/test/java/org/kitodo/selenium/testframework/Browser.java]: Please remove this debug helper ;)

File: Kitodo/src/test/java/org/kitodo/selenium/testframework/pages/ProjectsPage.java [hunk 6]
```diff
@@ -181,23 +181,29 @@ public int countListedRulesets() throws Exception {
...
         WebElement detailsTable = Browser.getDriver()
-                .findElement(By.id("projectsTabView:projectsTable:" + index + ":projectDetailTable"));
+                .findElement(By.id(PROJECTS_TABLE + ":" + index + ":projectDetailTable"));
         return getTableDataByColumn(detailsTable, 1);
     }
 
-    public List<String> getTemplateDetails() {
+    public List<String> getTemplateDetails() throws Exception {
         int index = triggerRowToggle(templatesTable, "First template");
+        Thread.sleep(5000);
```
Comment (henning-gerhardt) [inline Kitodo/src/test/java/org/kitodo/selenium/testframework/pages/ProjectsPage.java]: Why did you add this and a few lines below this call and only returning an empty list? I think this and similar line below could be removed.

**Comment by henning-gerhardt** [comment]
There are merge conflicts in

- Kitodo/src/main/java/org/goobi/mq/processors/CreateNewProcessProcessor.java
- Kitodo/src/main/java/org/kitodo/dto/TemplateDTO.java
- Kitodo/src/main/java/org/kitodo/services/data/TemplateService.java
- Kitodo/src/main/resources/messages/messages_de.properties
- Kitodo/src/main/resources/messages/messages_en.properties
- Kitodo/src/main/webapp/WEB-INF/templates/includes/projects/templateList.xhtml
- Kitodo/src/test/java/org/kitodo/selenium/AddingST.java
- Kitodo/src/test/java/org/kitodo/selenium/ListingST.java
- Kitodo/src/test/java/org/kitodo/selenium/testframework/pages/ProjectsPage.java 

Please fix / solve this conflicts.
